src/server/db_slice.h

```diff
@@ -69,6 +95,79 @@ class DbSlice {
   void operator=(const DbSlice&) = delete;
 
  public:
+  template <typename T> class IteratorT {
+   public:
+    IteratorT() = default;
+
+    IteratorT(T it, StringOrView key)
+        : it_(it), fiber_epoch_(util::fb2::FiberSwitchEpoch()), key_(std::move(key)) {
+    }
+
+    static IteratorT FromPrime(T it) {
+      std::string key;
+      it->first.GetString(&key);
+      return IteratorT(it, StringOrView::FromString(std::move(key)));
+    }
+
+    IteratorT(const IteratorT& o) = default;
+    IteratorT(IteratorT&& o) = default;
+    IteratorT& operator=(const IteratorT& o) = default;
+    IteratorT& operator=(IteratorT&& o) = default;
+
+    T GetInnerIt() {
+      LaunderIfNeeded();
+      return it_;
+    }
+
+    auto operator->() const {
+      LaunderIfNeeded();
+      return it_.operator->();
+    }
+
+    auto is_done() const {
+      LaunderIfNeeded();
+      return it_.is_done();
+    }
+
+    std::string_view key() const {
+      return key_.view();
+    }
+
+    auto IsOccupied() const {
+      LaunderIfNeeded();
+      return it_.IsOccupied();
+    }
+
+    bool IsValid() const {
+      LaunderIfNeeded();
+      return dfly::IsValid(it_);
+    }
```

Comment:
Instead of accesing it_ you can call just GetInnerIt(), so you don't have to call Launder manually

Reply:
Good idea!